src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3PayForm.tsx

```diff
@@ -59,16 +79,105 @@ export const V2V3PayForm = ({
   const canAddMoreStickers =
     (stickerUrls ?? []).length < ProjectPreferences.MAX_IMAGES_PAYMENT_MEMO
 
-  useEffect(
-    () => {
-      setCustomBeneficiaryEnabled(Boolean(form.getFieldValue('beneficiary')))
-    },
-    [], // eslint-disable-line react-hooks/exhaustive-deps
+  const reservedRate = fundingCycleMetadata?.reservedRate?.toNumber()
+
+  const receivedTickets = weightAmountPermyriad(
+    fundingCycle?.weight,
+    reservedRate,
+    weiAmount,
+    'payer',
   )
+
+  const tokenText = tokenSymbolText({
+    tokenSymbol,
+    plural: true,
+  })
+
+  const nftRewardTiers = useNftRewardTiersToMint()
+
   return (
     <>
       <Form form={form} layout="vertical" {...props}>
-        <Space direction="vertical" size="large">
+        <Space direction="vertical" size="large" className="w-full">
+          <div>
+            <Space className="w-full" direction="vertical" size="large">
+              <div className="flex justify-between">
+                <div className="font-bold">
+                  <Trans>Amount:</Trans>
+                </div>
+                <div>
+                  {formattedNum(usdAmount)}{' '}
+                  {V2V3CurrencyName(V2V3_CURRENCY_USD)} ({formatWad(weiAmount)}{' '}
+                  {V2V3CurrencyName(V2V3_CURRENCY_ETH)})
+                </div>
+              </div>
+
+              <div className="flex items-baseline justify-between">
+                <div className="font-bold">
+                  <Trans>Receive:</Trans>
+                </div>
+                <div className="text-right">
+                  <div className="inline-flex items-baseline">
+                    <Form.Item
+                      hidden={!customBeneficiaryEnabled}
+                      className="w-full"
+                      name="beneficiary"
+                      rules={[
+                        {
+                          validator: (_, value) => {
+                            if (!value || !isAddress(value)) {
+                              return Promise.reject('Address is required')
+                            }
+                            if (isZeroAddress(value)) {
+                              return Promise.reject('Cannot use zero address')
+                            }
+                            return Promise.resolve()
+                          },
+                          validateTrigger: 'onCreate',
+                          required: true,
+                        },
+                      ]}
+                    >
+                      <EthAddressInput />
+                    </Form.Item>
+
+                    {beneficiary && !customBeneficiaryEnabled && (
```

Comment:
we use a FormattedAddress to show the beneficiary while not editing. in this state the Form.Item is left in the dom and only `hidden`, so it doesn't interfere with the Form functionality